src/github.com/getlantern/flashlight/ui/service.go

```diff
@@ -0,0 +1,102 @@
+package ui
+
+import (
+	"encoding/json"
+	"sync"
+)
+
+type envelopeType struct {
+	Type string
+}
+
+type Service struct {
+	Name    string
+	In      chan []byte
+	Out     chan []byte
+	helloFn func(func([]byte) error) error
+}
+
+var (
+	mu               sync.Mutex
+	defaultUIChannel *UIChannel
+
+	out      = make(chan []byte, 10)
+	services = make(map[string]*Service)
+)
+
+func (s *Service) watch() {
+	// Watch for new messages and sent them to the combined output.
+	for b := range s.Out {
+		out <- b
+	}
+}
+
+func Register(name string, helloFn func(func([]byte) error) error) (*Service, error) {
```

Comment:
Yes, but in this case service name is always equal to the message type. Regarding the URL, as far as I know, the URL is just used to name a central location for websocket clients to locate a data provider, so, we can't really use it as a namespace unless opening another connection, that leaves us with the Type/Name as the only namespace.

I realized the `Name` property of Service is actually unused and is not mean to be changed, so I'll remove it to avoid confusion.